cJSON.c

```diff
@@ -1076,6 +1076,11 @@ CJSON_PUBLIC(cJSON *) cJSON_ParseWithOpts(const char *value, const char **return
     return NULL;
 }
 
+CJSON_PUBLIC(cJSON *) cJSON_ParseWithOpts(const char *value, const char **return_parse_end, cJSON_bool require_null_terminated)
+{
+    return cJSON_ParseWithLen(value, return_parse_end, require_null_terminated, strlen(value) + 1);
```

Comment:
Given that `cJSON_ParseWithLen` is supposed to work without the zero terminator, I wouldn't add the `+ 1` here. This would be an excellent test for if this really works without the zero terminator by using all of the existing tests.

Reply:
The problem was the "require_null_terminated" attribute.  Presumably I could make the +1 here conditional on whether that was set or not, or possibly check it in this function and then not pass it through (which in retrospect would be the better choice IMO.)